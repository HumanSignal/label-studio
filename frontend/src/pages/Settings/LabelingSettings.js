import { useCallback, useState } from 'react';
import { useHistory } from 'react-router';
import { useAPI } from '../../providers/ApiProvider';
import { useProject } from '../../providers/ProjectProvider';
import { ConfigPage } from '../CreateProject/Config/Config';

export const LabelingSettings = () => {
  const history = useHistory();
  const {project} = useProject();
  const [config, setConfig] = useState("");
  const api = useAPI();

  const onSave = useCallback(async () => {
    const res = await api.callApi("updateProjectRaw", {
      params: {
        pk: project.id,
      },
      body: {
        label_config: config,
      },
    });

    if (res.ok) {
      history.push(`/projects/${project.id}/data`);
      return true;
    }

    const error = await res.json();
    return error;
  }, [project, config]);

  if (!project.id) return null;

  return <ConfigPage config={project.label_config} project={project} onUpdate={setConfig} onSaveClick={onSave} />;
};

LabelingSettings.title = "Labeling Interface";
LabelingSettings.path = "/labeling";
