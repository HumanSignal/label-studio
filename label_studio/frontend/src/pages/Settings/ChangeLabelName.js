import { useEffect, useMemo, useState } from "react";
import { useHistory } from "react-router";
import { Button } from "../../components";
import { Label } from "../../components/Form";
import { confirm } from "../../components/Modal/Modal";
import { Space } from "../../components/Space/Space";
import { Spinner } from "../../components/Spinner/Spinner";
import { useAPI } from "../../providers/ApiProvider";
import { useProject } from "../../providers/ProjectProvider";
import axios from 'axios'
import getWebhookUrl from "../../webhooks";
import Swal from 'sweetalert2'

export const ChangeLabelName = () => {
  const webhook_url = getWebhookUrl();
  const {project} = useProject();
  const api = useAPI();
  const history = useHistory();
  const [selectedLabel, setSelectedLabel] = useState('');
  const [newLabel, setNewLabel] = useState('');

  const [availableLabels, setAvailableLabels] = useState([]);
  const [functionRunning, setFunctionRunning] = useState(false);
  useEffect(() => {
    console.log('re-rendering');
    if (project.id) {
      getLabels();
    }
  }, [project.id]);

  async function getLabels() {
    await axios
      .get(webhook_url + '/get_labels?id=' + project.id)
      .then((response) => {
        if (response.data.labels !== availableLabels && (response.data.labels instanceof Array)) {
          setAvailableLabels(response.data.labels);
        }

  })
}
  const handleSelect = (event) => {
    setSelectedLabel(event.target.value);
  };

  const handleChange = (event) => {
    setNewLabel(event.target.value);
  };

   const handleSubmit = async (event) => {
     event.preventDefault();
     setFunctionRunning(true);

    const updatedLabels = availableLabels.map((label) => {
      if (label === selectedLabel) {
        return newLabel;
      }
      return label;
    });
     console.log('Updated labels: ', updatedLabels);
     Swal.fire({
      title: 'Running',
      text: "Your label is being named, this may take some time!",
      icon: 'info',
    })
    await axios.post(webhook_url + '/change_label_name?id=' + project.id + '&current_name=' + selectedLabel + '&new_name=' + newLabel)
    .then((response) => {
      console.log(response);
      setFunctionRunning(false);
      setAvailableLabels(updatedLabels);
      Swal.fire({
        title: 'Success',
        text: "Your label has been renamed!",
        icon: 'success',
      })
  })
  };

  return (
    <div><h3>Change Label Name</h3>
    <form onSubmit={handleSubmit} style={{marginTop: 30}}>
      <label>
        Select a label:
        <select value={selectedLabel} onChange={handleSelect} style={{marginLeft: 10}}>
          <option value="" disabled>
            Select a label
          </option>
          {availableLabels?.map((label) => (
            <option key={label} value={label}>
              {label}
            </option>
          ))}
        </select>
      </label>
      <br />
      <br />
      <label>
        Change label to:
        <input style={{marginLeft: 10}} type="text" value={newLabel} onChange={handleChange} />
      </label>
      <br />
      <br />
      <button disabled={functionRunning} type="submit">Submit</button>
      </form> 
      </div>  
  );
};

ChangeLabelName.title = "Change Label Name";
ChangeLabelName.path = "/change-label-name";
