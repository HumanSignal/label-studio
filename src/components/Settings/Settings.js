import React from "react";
import { Modal, Checkbox, Tabs } from "antd";
import { observer } from "mobx-react";

export default observer(({ store }) => {
  return (
    <Modal
      visible={store.showingSettings}
      title="Settings"
      bodyStyle={{ paddingTop: "0" }}
      footer=""
      onCancel={store.toggleSettings}
    >
      <Tabs defaultActiveKey="1">
        <Tabs.TabPane tab="Hotkeys" key="1">
          <Checkbox
            value="Enable labeling hotkeys"
            defaultChecked={store.settings.enableHotkeys}
            onChange={() => {
              store.settings.toggleHotkeys();
            }}
          >
            Enable labeling hotkeys
          </Checkbox>
          <br />
          <Checkbox
            value="Show tooltips"
            defaultChecked={store.settings.enableTooltips}
            onChange={() => {
              store.settings.toggleTooltips();
            }}
          >
            Show tooltips
          </Checkbox>
        </Tabs.TabPane>
      </Tabs>
    </Modal>
  );
});
