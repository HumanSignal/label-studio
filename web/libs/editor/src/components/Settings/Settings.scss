%settings-title {
  font-family: Figtree, sans-serif;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.15px;
  text-align: left;
}


.settings__title {
  @extend %settings-title;
}

.settings {
  &__field + &__field {
    margin-top: 10px;
  }
}

.settings_newUI {
  .settings__field {
    display: flex;
    align-items: flex-start;
    cursor: pointer;
  }

  .settings__field + .settings__field {
    margin-top: 16px;
  }

  .settings__label {
    flex: 1;

    &__title {
      @extend %settings-title;

      align-items: center;
      display: flex;
      gap: 8px;
    }

    &__description {
      font-size: 14px;
      font-weight: 400;
      line-height: 20px;
      letter-spacing: 0.25px;
      text-align: left;
    }
  }
}

.keys {
  &__key {
    padding: 0 4px;
    height: 16px;
    display: inline-flex;
    align-items: center;
    background: var(--sand_0);
    border-radius: 2px;
    font-weight: bold;
    font-size: 12px;
    text-transform: uppercase;
    box-shadow: 0 0 0 1px rgb(0 0 0 / 10%), 0 1px 0 rgb(0 0 0 / 8%);
  }

  &__key + &__key {
    margin-left: 4px;
  }

  &__key-group {
    margin-left: -8px;
    margin-right: -8px;
    padding-left: 8px;
    padding-right: 8px;
  }

  &__key-group + &__key-group {
    margin-top: 5px;
    border-top: 2px solid var(--sand_200);
  }
}

.settings-modal {
  --modal-button-text-color-active: var(--primary_link);
  --modal-border-radius: 8px;
  --modal-padding: 24px;

  width: 100% !important;
  max-width: 568px !important;

  .ant-pagination-item-active {
    border-color: var(--primary_link);
  }

  .ant-pagination-item-active a {
    color: var(--primary_link);
  }

  :global(.ant-tabs-tab .ant-tabs-tab-btn) {
    @extend %settings-title;

    text-shadow: none;
  }

  :global(.ant-tabs-ink-bar) {
    background: var(--primary_link);
  }

  :global(.ant-modal-close) {
    top: 28px;
    right: var(--modal-padding);
  }

  :global(.ant-modal-close-icon) {
    display: none;
  }

  :global(.ant-modal-close-x) {
    height: 24px !important;
    width: 24px !important;
    line-height: 24px !important;
    color: var(--modal-button-text-color-active);
  }

  :global(.ant-tabs-tab) {
    padding: calc(var(--modal-padding) / 4) var(--modal-padding) !important;

    :hover {
      color: var(--primary_link);
    }
  }

  :global(.ant-tabs-tab + .ant-tabs-tab) {
    margin-left: 0 !important;
  }

  :global(.ant-tabs-tab.ant-tabs-tab-active .ant-tabs-tab-btn) {
    color: var(--modal-button-text-color-active) !important;
  }

  :global(.ant-modal-content) {
    border-radius: var(--modal-border-radius) !important;
  }

  :global(.ant-modal-body) {
    padding: 0 var(--modal-padding) var(--modal-padding) !important;
  }

  :global(.ant-modal-header) {
    border-radius: var(--modal-border-radius) var(--modal-border-radius) 0 0 !important;
    padding: var(--modal-padding) !important;
    border: none !important;
  }

  :global(.ant-modal-title) {
    font-size: 24px;
    font-weight: 500;
    line-height: 32px;
    letter-spacing: 0;
    text-align: left;
  }

  .settings-tag {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    padding-inline: 4px;
    color: var(--grape_700);
    background-color: var(--grape_0);
    border-radius: 2px;
    height: 20px;
    font-size: 11px;
    font-weight: 500;
    line-height: 16px;
    letter-spacing: 0.5px;
    text-align: center;
  }
}
