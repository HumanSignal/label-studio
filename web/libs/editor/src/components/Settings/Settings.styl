
$settings__title
  color #1F1F1F
  font-family Roboto
  font-size 16px
  font-weight 500
  line-height 24px
  letter-spacing 0.15px
  text-align left

.settings
  &__field + &__field
    margin-top 10px

.settings_newUI
  .settings__field
    display flex
    align-items flex-start
    cursor pointer
  .settings__field + .settings__field
    margin-top 16px
  .settings__label
    flex 1
    &__title
      @extends $settings__title
      align-items center
      display flex
      gap 8px
    &__description
      font-size 14px
      font-weight 400
      line-height 20px
      letter-spacing 0.25px
      text-align left

.keys
  &__key
    padding 0 4px
    height 16px
    display inline-flex
    align-items center
    background #FFFFFF
    border-radius 2px
    font-weight bold
    font-size 12px
    text-transform uppercase
    box-shadow 0px 0px 0px 1px rgba(0, 0, 0, 0.1), 0px 1px 0px rgba(0, 0, 0, 0.08)

  &__key + &__key
    margin-left 4px

  &__key-group
    margin-left -8px
    margin-right -8px
    padding-left 8px
    padding-right 8px

  &__key-group + &__key-group
    margin-top 5px
    border-top 2px solid rgba(#000, 0.05)

.settings-modal
  --modal-button-text-color-active #40A9FF
  --toggle-color #1890FF
  --modal-border-radius 8px
  --modal-padding 24px

  width 100% !important
  max-width 568px !important

  :global(.ant-tabs-tab .ant-tabs-tab-btn)
    @extends $settings__title
    text-shadow none
  :global(.ant-modal-close)
    top: 28px
    right: var(--modal-padding)
  :global(.ant-modal-close-icon)
    display none
  :global(.ant-modal-close-x)
    height 24px !important
    width 24px !important
    line-height 24px !important
    color var(--modal-button-text-color-active)
  :global(.ant-tabs-tab)
    padding calc(var(--modal-padding) / 4) var(--modal-padding) !important
  :global(.ant-tabs-tab + .ant-tabs-tab)
    margin-left 0 !important
  :global(.ant-tabs-tab.ant-tabs-tab-active .ant-tabs-tab-btn)
    color var(--modal-button-text-color-active) !important
  :global(.ant-modal-content)
    border-radius var(--modal-border-radius) !important
  :global(.ant-modal-body)
    padding 0 var(--modal-padding) var(--modal-padding) !important
  :global(.ant-modal-header)
    border-radius: var(--modal-border-radius) var(--modal-border-radius) 0 0 !important;
    padding var(--modal-padding) !important
    border none !important
  :global(.ant-modal-title)
    color #1F1F1F
    font-family Roboto
    font-size 24px
    font-weight 400
    line-height 32px
    letter-spacing 0
    text-align left

  .settings-tag
    display inline-flex
    align-items center
    justify-content center
    padding-inline 4px
    color #030852
    background-color #D6E4FF
    border-radius 2px
    height 20px
    font-size 11px
    font-weight 500
    line-height 16px
    letter-spacing 0.5px
    text-align center
