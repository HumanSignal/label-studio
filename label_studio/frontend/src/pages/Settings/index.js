import React from 'react';
import { SidebarMenu } from '../../components/SidebarMenu/SidebarMenu';
import { WebhookPage } from '../WebhookPage/WebhookPage';
import { DangerZone } from './DangerZone';
import { GeneralSettings } from './GeneralSettings';
import { InstructionsSettings } from './InstructionsSettings';
import { LabelingSettings } from './LabelingSettings';
import { MachineLearningSettings } from './MachineLearningSettings/MachineLearningSettings';
import { StorageSettings } from './StorageSettings/StorageSettings';


export const MenuLayout = ({children, ...routeProps}) => {
  return (
    <SidebarMenu
      menuItems={[
        GeneralSettings,
        LabelingSettings,
        InstructionsSettings,
        MachineLearningSettings,
        StorageSettings,
        WebhookPage,
        DangerZone,
      ]}
      path={routeProps.match.url}
      children={children}
    />
  );
};

export const SettingsPage = {
  title: "Settings",
  path: "/settings",
  exact: true,
  layout: MenuLayout,
  component: GeneralSettings,
  pages: {
    InstructionsSettings,
    LabelingSettings,
    MachineLearningSettings,
    StorageSettings,
    WebhookPage,
    DangerZone,
  },
};
