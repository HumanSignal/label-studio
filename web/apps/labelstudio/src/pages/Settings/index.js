import React from 'react';
import { SidebarMenu } from '../../components/SidebarMenu/SidebarMenu';
import { WebhookPage } from '../WebhookPage/WebhookPage';
import { DangerZone } from './DangerZone';
import { GeneralSettings } from './GeneralSettings';
import { AnnotationSettings } from './AnnotationSettings';
import { LabelingSettings } from './LabelingSettings';
import { MachineLearningSettings } from './MachineLearningSettings/MachineLearningSettings';
import { PredictionsSettings } from './PredictionsSettings/PredictionsSettings';
import { StorageSettings } from './StorageSettings/StorageSettings';
import {
  isInLicense,
  LF_CLOUD_STORAGE_FOR_MANAGERS
} from '../../utils/license-flags';

const isAllowCloudStorage = !isInLicense(LF_CLOUD_STORAGE_FOR_MANAGERS);

export const MenuLayout = ({ children, ...routeProps }) => {
  return (
    <SidebarMenu
      menuItems={[
        GeneralSettings,
        LabelingSettings,
        AnnotationSettings,
        MachineLearningSettings,
        PredictionsSettings,
        isAllowCloudStorage && StorageSettings,
        WebhookPage,
        DangerZone,
      ].filter(Boolean)}
      path={routeProps.match.url}
      children={children}
    />
  );
};

const pages = {
  AnnotationSettings,
  LabelingSettings,
  MachineLearningSettings,
  PredictionsSettings,
  WebhookPage,
  DangerZone,
};

isAllowCloudStorage && (pages.StorageSettings = StorageSettings);

export const SettingsPage = {
  title: 'Settings',
  path: '/settings',
  exact: true,
  layout: MenuLayout,
  component: GeneralSettings,
  pages,
};
