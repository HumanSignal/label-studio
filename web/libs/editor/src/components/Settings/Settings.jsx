import React, { useMemo } from 'react';
import { Checkbox, Modal, Table, Tabs } from 'antd';
import { observer } from 'mobx-react';

import { Hotkey } from '../../core/Hotkey';

import './Settings.styl';
import { Block, Elem } from '../../utils/bem';
import { triggerResizeEvent } from '../../utils/utilities';

import EditorSettings from '../../core/settings/editorsettings';
import * as TagSettings from './TagSettings';
import { LsClose } from '../../assets/icons';
import Toggle from '../../common/Toggle/Toggle';
import { FF_DEV_3873, isFF } from '../../utils/feature-flags';

const HotkeysDescription = () => {
  const columns = [
    { title: 'Shortcut', dataIndex: 'combo', key: 'combo' },
    { title: 'Description', dataIndex: 'descr', key: 'descr' },
  ];

  const keyNamespaces = Hotkey.namespaces();

  const getData = (descr) => Object.keys(descr)
    .filter(k => descr[k])
    .map(k => ({
      key: k,
      combo: k.split(',').map(keyGroup => {
        return (
          <Elem name="key-group" key={keyGroup}>
            {keyGroup.trim().split('+').map((k) => <Elem tag="kbd" name="key" key={k}>{k}</Elem>)}
          </Elem>
        );
      }),
      descr: descr[k],
    }));

  return (
    <Block name="keys">
      <Tabs size="small">
        {Object.entries(keyNamespaces).map(([ns, data]) => {
          if (Object.keys(data.descriptions).length === 0) {
            return null;
          } else {
            return (
              <Tabs.TabPane key={ns} tab={data.description ?? ns}>
                <Table columns={columns} dataSource={getData(data.descriptions)} size="small" />
              </Tabs.TabPane>
            );
          }
        })}
      </Tabs>
    </Block>
  );
};


const newUI = isFF(FF_DEV_3873) ? { newUI: true } : {};

const editorSettingsKeys = Object.keys(EditorSettings);

if (isFF(FF_DEV_3873)) {
  const enableTooltipsIndex = editorSettingsKeys.findIndex(key => key === 'enableTooltips');
  const enableLabelTooltipsIndex = editorSettingsKeys.findIndex(key => key === 'enableLabelTooltips');

  // swap these in the array
  const tmp = editorSettingsKeys[enableTooltipsIndex];

  editorSettingsKeys[enableTooltipsIndex] = editorSettingsKeys[enableLabelTooltipsIndex];
  editorSettingsKeys[enableLabelTooltipsIndex] = tmp;
}

const SettingsTag = ({ children }) => {
  return (
    <Block name="settings-tag">{children}</Block>
  );
};

const GeneralSettings = observer(({ store }) => {
  return (
    <Block name="settings" mod={newUI}>
      {editorSettingsKeys.map((obj, index) => {
        return (
          <Elem name="field" tag="label" key={index}>
            {isFF(FF_DEV_3873) ? (
              <>
                <Block name="settings__label">
                  <Elem name="title">
                    {EditorSettings[obj].newUI.title}
                    {EditorSettings[obj].newUI.tags?.split(',').map((tag) => (<SettingsTag key={tag}>{tag}</SettingsTag>))}
                  </Elem>
                  <Block name="description">
                    {EditorSettings[obj].newUI.description}
                  </Block>
                </Block>
                <Toggle
                  key={index}
                  checked={store.settings[obj]}
                  onChange={store.settings[EditorSettings[obj].onChangeEvent]}
                  description={EditorSettings[obj].description}
                />
              </>
            ) : (
              <>
                <Checkbox
                  key={index}
                  checked={store.settings[obj]}
                  onChange={store.settings[EditorSettings[obj].onChangeEvent]}
                >
                  {EditorSettings[obj].description}
                </Checkbox>
                <br />
              </>
            )}
          </Elem>
        );
      })}
    </Block>
  );
});

const LayoutSettings = observer(({ store }) => {
  return (
    <Block name="settings" mod={newUI}>
      <Elem name="field">
        <Checkbox
          checked={store.settings.bottomSidePanel}
          onChange={() => {
            store.settings.toggleBottomSP();
            setTimeout(triggerResizeEvent);
          }}
        >
          Move sidepanel to the bottom
        </Checkbox>
      </Elem>

      <Elem name="field">
        <Checkbox checked={store.settings.displayLabelsByDefault} onChange={store.settings.toggleSidepanelModel}>
          Display Labels by default in Results panel
        </Checkbox>
      </Elem>

      <Elem name="field">
        <Checkbox
          value="Show Annotations panel"
          defaultChecked={store.settings.showAnnotationsPanel}
          onChange={() => {
            store.settings.toggleAnnotationsPanel();
          }}
        >
          Show Annotations panel
        </Checkbox>
      </Elem>

      <Elem name="field">
        <Checkbox
          value="Show Predictions panel"
          defaultChecked={store.settings.showPredictionsPanel}
          onChange={() => {
            store.settings.togglePredictionsPanel();
          }}
        >
          Show Predictions panel
        </Checkbox>
      </Elem>

      {/* Saved for future use */}
      {/* <Elem name="field">
        <Checkbox
          value="Show image in fullsize"
          defaultChecked={store.settings.imageFullSize}
          onChange={() => {
            store.settings.toggleImageFS();
          }}
        >
          Show image in fullsize
        </Checkbox>
      </Elem> */}
    </Block>
  );
});

const Settings = {
  General: { name: 'General', component: GeneralSettings },
  Hotkeys: { name: 'Hotkeys', component: HotkeysDescription },
};

if (!isFF(FF_DEV_3873)) {
  Settings.Layout = { name: 'Layout', component: LayoutSettings };
}

const DEFAULT_ACTIVE = Object.keys(Settings)[0];

const DEFAULT_MODAL_SETTINGS = isFF(FF_DEV_3873) ? {
  name: 'settings-modal',
  title: 'Labeling Interface Settings',
  closeIcon: <LsClose />,
} : {
  name: 'settings-modal-old',
  title: 'Settings',
  bodyStyle: { paddingTop: '0' },
};

export default observer(({ store }) => {
  const availableSettings = useMemo(() => {
    const availableTags = Object.values(store.annotationStore.names.toJSON());
    const settingsScreens = Object.values(TagSettings);

    return availableTags.reduce((res, tagName) => {
      const tagType = store.annotationStore.names.get(tagName).type;
      const settings = settingsScreens.find(({ tagName }) => tagName.toLowerCase() === tagType.toLowerCase());

      if (settings) res.push(settings);

      return res;
    }, []);
  }, []);

  return (
    <Block
      tag={Modal}
      visible={store.showingSettings}
      onCancel={store.toggleSettings}
      footer=""
      {...DEFAULT_MODAL_SETTINGS}
    >
      <Tabs defaultActiveKey={DEFAULT_ACTIVE}>
        {Object.entries(Settings).map(([key, { name, component }]) => (
          <Tabs.TabPane tab={name} key={key}>
            {React.createElement(component, { store })}
          </Tabs.TabPane>
        ))}
        {availableSettings.map((Page) => (
          <Tabs.TabPane tab={Page.title} key={Page.tagName}>
            <Page store={store} />
          </Tabs.TabPane>
        ))}
      </Tabs>
    </Block>
  );
});
