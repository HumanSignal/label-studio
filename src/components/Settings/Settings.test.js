import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import Settings from "./Settings";

describe("Settings Component", () => {
  it("Should render correctly", () => {
    const output = shallow(<Settings store={{ showingSettings: true, settings: { enableHotkeys: true } }} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
