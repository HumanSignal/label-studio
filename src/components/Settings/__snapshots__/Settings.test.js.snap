// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component Should render correctly 1`] = `
<Modal
  confirmLoading={false}
  footer=""
  maskTransitionName="fade"
  okType="primary"
  title="Hotkeys"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Checkbox
    defaultChecked={true}
    indeterminate={false}
    onChange={[Function]}
    value="Enable labeling hotkeys"
  >
    Enable labeling hotkeys
  </Checkbox>
  <br />
  <Checkbox
    indeterminate={false}
    onChange={[Function]}
    value="Show tooltips"
  >
    Show tooltips
  </Checkbox>
</Modal>
`;
